pybossa/cache/projects.py

```diff
@@ -81,8 +82,22 @@ def browse_tasks(project_id, args, filter_user_prefs=False, user_id=None):
             MAX(finish_time) as ft FROM task_run
             WHERE project_id=:project_id GROUP BY task_id) AS log_counts
             ON task.id=log_counts.task_id
-            WHERE task.project_id=:project_id""" + filters + \
+            WHERE task.state='ongoing'
```

Comment:
The where clause should exclude tasks that the current worker has worked before, `task.state=ongoing` does not filter out those have been worked by the current worker.